src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,182 @@
+
+    public static void printDividerLine() {
+        System.out.println("\t_____________________________________________________________________________");
+    }
+
+    public void printAddedTasked() {
+        printDividerLine();
+        System.out.println("\t Got it. I've added this task: ");
+        System.out.println("\t   " + tasks.get(Task.getNumberOfTasks() - 1));
+        System.out.println("\t Now you have " + Task.getNumberOfTasks() + " tasks in the list.");
+        printDividerLine();
+    }
+
+    public void printIncorrectInput(boolean isEmptyDescription, String keyword) {
+        printDividerLine();
+        if (keyword.equalsIgnoreCase("todo")) {
+            System.out.println("\t ☹ OOPS!!! The description of a todo cannot be empty.");
+        } else if (keyword.equalsIgnoreCase("deadline")) {
+            if (isEmptyDescription) {
+                System.out.println("\t ☹ OOPS!!! The description of a deadline cannot be empty.");
+            } else {
+                System.out.println("\t ☹ OOPS!!! The description of a deadline cannot be in the wrong format.");
+            }
+        } else {
+            if (isEmptyDescription) {
+                System.out.println("\t ☹ OOPS!!! The description of an event cannot be empty.");
+            } else {
+                System.out.println("\t ☹ OOPS!!! The description of an event cannot be in the wrong format.");
+            }
+        }
+        printDividerLine();
+    }
+
+    public void addTask(String line) {
+        String[] keywords = line.split(" ", 2);
+        boolean hasNoException = false;
+        boolean isEmptyDescription = true;
+
+        if (keywords[INDEX_OF_KEYWORD_OF_TASK].equalsIgnoreCase("todo")) {
+            try {
+                tasks.add(new Todo(keywords[1]));
+                hasNoException = true;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printIncorrectInput(false, keywords[INDEX_OF_KEYWORD_OF_TASK]);
+            }
+        } else if (keywords[INDEX_OF_KEYWORD_OF_TASK].equalsIgnoreCase("deadline")) {
+            try {
+                String[] details = keywords[1].split(" /by ");
+                isEmptyDescription = false;
+                tasks.add(new Deadline(details[0], details[1]));
+                hasNoException = true;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printIncorrectInput(isEmptyDescription, keywords[INDEX_OF_KEYWORD_OF_TASK]);
+            }
+        } else if (keywords[INDEX_OF_KEYWORD_OF_TASK].equalsIgnoreCase("event")) {
+            try {
+                String[] details = keywords[1].split(" /at ");
+                isEmptyDescription = false;
+                tasks.add(new Event(details[0], details[1]));
+                hasNoException = true;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printIncorrectInput(isEmptyDescription, keywords[INDEX_OF_KEYWORD_OF_TASK]);
+            }
+        }
+        if (hasNoException) {
```

Comment:
Instead of using a boolean to see if an exception has occurred, you may consider throwing the exception back to the caller method and let the calling method handle the exception. In that way, you can confidently printAddedTasked() and updateFile(). Because if an exception had occurred, an exception will be thrown from the method and this part of the code will not be reached. Just a suggestion :)